packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2428,6 +2446,22 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     );
   }
 
+  /// Launch a web search on the current selection, as in the "Search Web" edit menu button on iOS.
+  /// Currently this is only implemented for iOS.
+  /// Throws an error if the selection is empty or collapsed.
+  Future<void> searchWebForSelection(SelectionChangedCause cause) async {
+    assert(!widget.obscureText);
+
+    final String text = textEditingValue.selection.textInside(textEditingValue.text);
+    if (widget.obscureText || text.isEmpty) {
```

Comment:
Nit: Maybe do the `widget.obscureText` case before doing textInside to avoid the unnecessary operation.  Not a big deal though, readability is more important.